pyogrio/geopandas.py

```diff
@@ -306,7 +306,22 @@ def write_dataframe(
     fields = [c for c in df.columns if not c == geometry_column]
 
     # TODO: may need to fill in pd.NA, etc
-    field_data = [df[f].values for f in fields]
+    field_data = []
+    field_mask = []
+    for name in fields:
+        col = df[name].values
+        if isinstance(col, pd.api.extensions.ExtensionArray):
+            from pandas.arrays import IntegerArray, FloatingArray, BooleanArray
+
+            if isinstance(col, (IntegerArray, FloatingArray, BooleanArray)):
+                field_data.append(col._data)
```

Comment:
This feels a little bit like reaching into the non-public API, which I thought was also marked as experimental / subject to change.  I'm assuming you did this due to the overhead of `.isna()`?

Reply:
Yes, this is indeed private API, but there is also not really another option right now to get the actual data (`isna` could be used to get the mask (although that involves doing an unnecessary copy), but for getting the data there is no other way (eg np.asarray(col) would convert the ints to floats if there are missing values). 
See some discussion at https://github.com/pandas-dev/pandas/issues/34873#issuecomment-652633878 which concluded that for now other libraries could use those attributes.